relay-metrics/src/bucket.rs

```diff
@@ -556,10 +565,105 @@ impl From<MetricValue> for BucketValue {
     }
 }
 
-/// Error returned when parsing or serializing a [`Bucket`].
-#[derive(Debug, Error)]
-#[error("failed to parse metric bucket")]
-pub struct ParseBucketError(#[source] serde_json::Error);
+/// Parses a list of counter values separated by colons and sums them up.
+fn parse_counter(string: &str) -> Option<CounterType> {
+    let mut sum = CounterType::default();
+    for component in string.split(':') {
+        sum += component.parse::<CounterType>().ok()?;
+    }
+    Some(sum)
+}
+
+/// Parses a distribution from a list of floating point values separated by colons.
+fn parse_distribution(string: &str) -> Option<DistributionValue> {
+    let mut dist = DistributionValue::default();
+    for component in string.split(':') {
```

Comment:
nit: Should we define a constant for `:`?

```suggestion
    for component in string.split(VALUE_SEPARATOR) {
```